### PR TITLE
CI: Compatibility Testing

### PR DESCRIPTION
**Pull request by riandyrn**
## Summary
- Add compatibility testing for windows
- Tested, you can see [here](https://github.com/riandyrn/otelchi/actions/runs/8477186545)

### CODE REVIEW

**Review by ilhamsyahids — APPROVED** [review]
LGTM 🚀

File: .github/workflows/compatibility-test.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  pull_request_review:
+    types: [submitted]
+
+jobs:
+  compatibility-test:
+    if: github.event.review.state == 'APPROVED'
+    strategy:
+      matrix:
+        go-version: ["1.15", "1.16", "1.17"]
+        os: [windows-latest]
```
[the remark this answers is not in the document]
Comment (riandyrn) [reply .github/workflows/compatibility-test.yaml]: Ok, I've updated it in [1fa7d4a](https://github.com/riandyrn/otelchi/pull/32/commits/1fa7d4a7921917db321ec322f08f9eee9ef61792).

File: .github/workflows/compatibility-test.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('**/go.sum') }}
+    - name: Run build
+      env:
+        GOARCH: ${{ matrix.arch }}
+      run: make go-build
+    - name: Run tests
+      env:
+        GOARCH: ${{ matrix.arch }}
+      run: make go-test
```
[the remark this answers is not in the document]
Comment (riandyrn) [reply .github/workflows/compatibility-test.yaml]: Ok, I've updated it in [c0c21df](https://github.com/riandyrn/otelchi/pull/32/commits/c0c21dff0507e5967532809611430ed010b3a81b).

**Comment by riandyrn** [comment]
Thanks a lot for the contributions, @ilhamsyahids!

**Pull request merged**